### PR TITLE
Refactor php 7.4

### PR DESCRIPTION
**Pull request by CarlSchwan**

### CODE REVIEW

**Review by come-nc — APPROVED** [review]

**Review by nickvergessen — CHANGES_REQUESTED** [review]
Installing fails on 7.4

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,65 +1,268 @@
...
-      working-directory: nextcloud/apps/recommendations
-      run: composer install
-    - name: Install app
-      run: php -f nextcloud/occ app:enable recommendations
-    - name: Install app
-      run: php -f nextcloud/occ app:enable recommendations
-    - name: Run a smoke test
-      run: php -f nextcloud/occ files:recommendations:recommend admin
+      - name: Summary status
+        run: if ${{ needs.oci.result != 'success' and needs.sqlite.result != 'success' and needs.mysql.result != 'success' and needs.pgsql.result != 'success'}}; then exit 1; fi
```
Comment (nickvergessen) [inline .github/workflows/test.yml:268]: Invalid workflow file: .github/workflows/test.yml#L268
The workflow is not valid. .github/workflows/test.yml (Line: 268, Col: 14): Unexpected symbol: 'and'. Located at position 31 within expression: needs.oci

**Comment by CarlSchwan** [comment]
> Installing fails on 7.4

works for me with both 7.4 and 8.1 locally

**Comment by ChristophWurst** [comment]
> @ChristophWurst I ended up updating composer and the ci jobs. The list of required ci checks needs to be changed in the github setting.

You have permission to adjust the requried checks :v:

**Review by ChristophWurst — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**